charts/gsp-cluster/templates/02-gsp-system/harbor-databases.yaml

```diff
@@ -0,0 +1,76 @@
+---
+# a duplicate of what gets created by the harbor chart
+# but as there's no reliable way to reference that secret
+# by name it is duplicated here based off the same entry in
+# the values.yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ .Release.Name }}-harbor-db-password
+  namespace: {{ .Release.Namespace }}
+data:
+  password: {{ .Values.harbor.database.external.password | b64enc | quote }}
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ .Release.Name }}-ensure-harbor-databases
+  namespace: {{ .Release.Namespace }}
+data:
+  ensure-databases.sql: |
+    --- based on https://stackoverflow.com/a/18389184
+    SELECT 'CREATE DATABASE {{ .Values.harbor.database.external.coreDatabase }}'
+    WHERE NOT EXISTS (SELECT FROM pg_database WHERE datname = '{{ .Values.harbor.database.external.coreDatabase }}')
+    UNION
+    SELECT 'CREATE DATABASE {{ .Values.harbor.database.external.clairDatabase }}'
+    WHERE NOT EXISTS (SELECT FROM pg_database WHERE datname = '{{ .Values.harbor.database.external.clairDatabase }}')
+    UNION
+    SELECT 'CREATE DATABASE {{ .Values.harbor.database.external.notaryServerDatabase }}'
+    WHERE NOT EXISTS (SELECT FROM pg_database WHERE datname = '{{ .Values.harbor.database.external.notaryServerDatabase }}')
+    UNION
+    SELECT 'CREATE DATABASE {{ .Values.harbor.database.external.notarySignerDatabase }}'
+    WHERE NOT EXISTS (SELECT FROM pg_database WHERE datname = '{{ .Values.harbor.database.external.notarySignerDatabase }}')
+    \gexec
+---
+# Based in part on
+# https://medium.com/faun/create-second-postgresql-db-in-the-pod-with-kubernetes-job-1b2f319e7c1b
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ .Release.Name }}-ensure-harbor-databases
+  namespace: {{ .Release.Namespace }}
+spec:
+  template:
+    spec:
+      containers:
+      - name: psql
+        image: postgres:10
```

Comment:
can we use a digest for this image instead of the label `10` to pin it

Reply:
Updated.